### ISSUE TITLE
Code action: add suggested imports

### ISSUE THREAD

**Issue by hsenag** [haskell/ghcide#46]
This compiler diagnostic could have the obvious code action to add the imported symbol.
```
C:\msys64\home\ganesh\temp\x\Bar.hs:5:9: error:
    * Variable not in scope: thething3
    * Perhaps you want to add `thething3' to the import list
      in the import of `Foo'
      (C:\msys64\home\ganesh\temp\x\Bar.hs:3:1-17).
```
I think this is probably fairly easy, but as with #45, it will require a bit of infrastructure to understand the structure of import lists to be able to edit them. (And it might be hard to preserve the "right" formatting.)

In some cases there's also a suggestion to rename the reference to something that's already in scope (which was implemented in #40) and ideally in that case both code actions would be offered.

**Comment by pepeiborra** [comment]
I would like to contribute a fix for this, but it's very unclear how to parse and extend the imports. The obvious approach seems to be reusing the GHC parse tree, modifying it with the extra import, and then pretty printing and generating a diff for LSP. Is this a reasonable plan? Obviously this assumes that the GHC parse tree respects comments and whitespace, which may or may not be the case.

**Comment by cocreature** [comment]
As far as I know, pretty printing the GHC AST messes up formatting pretty badly. If you want to preserve formatting you probably have to use [ghc-exactprint](https://hackage.haskell.org/package/ghc-exactprint). Alternatively, you can probably do something more crude like inserting a line after the last import or something like that.

**Comment by hsenag** [comment]
I had in mind something crude with just inserting the symbol before the last closing bracket.

**Comment by pepeiborra** [comment]
My first stab at this is ready: https://github.com/pepeiborra/ghcide/commit/9b5331d57eeef62d8008ff59f940d3171b71ee85

I'll send a merge request once my other MR for #45 is merged.

**Comment by pepeiborra** [comment]
This can probably be closed now.
